instructor/process_response.py

```diff
@@ -249,6 +250,15 @@ def handle_response_model(
             Mode.warn_mode_functions_deprecation()
             new_kwargs["functions"] = [response_model.openai_schema]
             new_kwargs["function_call"] = {"name": response_model.openai_schema["name"]}
+        elif mode == Mode.TOOLS_STRICT:
+            response_model_schema = pydantic_function_tool(response_model)
```

Comment:
This functionality here uses the OpenAI Function tool. we set the strict to True to configure the Structured Outputs mode

```
from openai import pydantic_function_tool
from pydantic import BaseModel


class User(BaseModel):
    name: str
    age: int


class Users(BaseModel):
    users: list[User]


print(pydantic_function_tool(Users))

# {
#     "type": "function",
#     "function": {
#         "name": "Users",
#         "strict": False,
#         "parameters": {
#             "$defs": {
#                 "User": {
#                     "properties": {
#                         "name": {"title": "Name", "type": "string"},
#                         "age": {"title": "Age", "type": "integer"},
#                     },
#                     "required": ["name", "age"],
#                     "title": "User",
#                     "type": "object",
#                     "additionalProperties": False,
#                 }
#             },
#             "properties": {
#                 "users": {
#                     "items": {"$ref": "#/$defs/User"},
#                     "title": "Users",
#                     "type": "array",
#                 }
#             },
#             "required": ["users"],
#             "title": "Users",
#             "type": "object",
#             "additionalProperties": False,
#         },
#     },
# }

```